Def 10 remove join from get order request

shaguahill opened the pull request:
1. Created new query to get photos to order and rewrote method getOrderByIdInStatusNew.
2. Added constructor to add photos to Order
3. Added parameter to application.properties allowMultiQueries=true

shaguahill committed:
[def-10] Remove join from get order request:
1. Created new query to get photos to order and rewrote method getOrderByIdInStatusNew.
2. Added constructor to add photos to Order
3. Added parameter to application.properties allowMultiQueries=true

shaguahill committed:
[def-10] Remove join from get order request(Small fix):
Deleted photos

Brutt reviewed (commented):
Maybe it would be better to use ResultSetExtractor (and JdbcTemplate) with LEFT JOIN instead of execute multiple queries.

shaguahill commented:
> Maybe it would be better to use ResultSetExtractor (and JdbcTemplate) with LEFT JOIN instead of execute multiple queries.

Isn`t it a Spring feature?

Brutt commented:
> > Maybe it would be better to use ResultSetExtractor (and JdbcTemplate) with LEFT JOIN instead of execute multiple queries.
> 
> Isn`t it a Spring feature?

Yeah, you are right.

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -144,13 +144,34 @@ public JdbcOrderDao(DataSource dataSource) {
+            return order;
 
         } catch (SQLException e) {
             log.error("Get order by id:{} in status NEW error", id);
```
а сама ошибка получается не логируется?

shaguahill replied:
Добавлено

shaguahill committed:
[def-10] Remove join from get order request(review fix):
Added exception message to log

Rooman commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -144,16 +144,37 @@ public JdbcOrderDao(DataSource dataSource) {
+                    while (photoResultSet.next()) {
+                        photoSources.add(photoResultSet.getString("source"));
+                    }
+                    if (photoSources.size() != 0) {
```
не понимаю зачем эта проверка? 
Почему нельзя подсетить в сущность, или в билдер фотки. Если их нету то будет подсечена пустая коллекция. 
А то получается мы
1. Лишний объект ордера создаем
2. Лишний афбричный метод по создаю ордера с картинками имеем

shaguahill replied:
Созданы:
Маппер для сбора Ордера с фото
Маппер для получения фото сорсов с резалтсета (В дальнейшем будет использоватся для сбора новой ентити Image/Photo)

Rooman commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -144,16 +144,37 @@ public JdbcOrderDao(DataSource dataSource) {
 
         } catch (SQLException e) {
-            log.error("Get order by id:{} in status NEW error", id);
+            log.error("Get order by id:{} in status NEW error", id, e);
```
пробел между : и аргументом

shaguahill replied:
Исправил

Rooman commented on src/main/java/com/photostudio/entity/order/Order.java:
```diff
@@ -19,4 +18,17 @@
     private String comment;
     private List<String> photoSources;
+
+    private Order(Order order, List<String> sources) {
```
смущаем этот констурктор... Предлагаю просто создавать ордер черех билдер или сеттеры.

Rooman replied:
а то у нас тут запутанная логика на констуркторах повязывается

shaguahill replied:
Конструктор убран, ордер собирается через билдер

shaguahill committed:
[def-10] Remove join from get order request(review fix):
Added mapper for order with photos
Added mapper for getting photo sources (will change to mapRow Image entity)
CHanged method getOrderByIdInStatusNew
delete private constructor and method from Order entity

shaguahill committed:
[def-10] Remove join from get order request(review fix):
Added exception when second resultSet == null

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -143,19 +145,38 @@ public JdbcOrderDao(DataSource dataSource) {
+            if (!preparedStatement.getMoreResults()) {
+                log.error("Get more resultSet for order with id: {} error", id);
+                throw new RuntimeException("Get more resultSet for order with id: " + id + " error");
+            }
```
т.е. если нет фото, то будет ошибка? был же left join? а тесты для этой функции есть?

shaguahill replied:
Нет, если по какой-то причине у нас не отработает второй запрос в мультизапросе и мы не получим второй резалтСет - тогда будет ошибка.
Если, чисто теоретически, второй запрос отработает и не вернет результатов - резалтСет все равно создается, но пустой.
Если фото не будет, мы просто добавим пустой лист.

shaguahill committed:
Merge branch 'develop' into def-10-remove-join-from-get-order-request

Rooman committed:
[def-10] REfactoring photo-sources mapping
